src/main/groovy/wooga/gradle/unity/utils/internal/ProjectSettings.groovy

```diff
@@ -18,52 +18,16 @@
 
 package wooga.gradle.unity.utils.internal
 
-import org.gradle.api.logging.Logger
-import org.gradle.api.logging.Logging
-import org.yaml.snakeyaml.Yaml
+import groovy.transform.InheritConstructors
 
-class ProjectSettings {
-
-    static Logger logger = Logging.getLogger(ProjectSettings)
-    private def content
-
-    ProjectSettings(File projectSettingsFile) {
-        this(projectSettingsFile.text)
-    }
-
-    ProjectSettings(String templateContent) {
-        try {
-            Yaml parser = new Yaml()
-            content = parser.load(stripUnityInstructions(templateContent))
-        }
-        catch (Exception e) {
-            logger.warn("Project Settings file not parsable. Please make sure it's not set to binary.")
-            content = null
-        }
-    }
+@InheritConstructors
+class ProjectSettings extends UnityAssetFile {
 
     boolean getPlayModeTestRunnerEnabled() {
-        isValidSettingsObject() && content['PlayerSettings']['playModeTestRunnerEnabled'] && content['PlayerSettings']['playModeTestRunnerEnabled'] == 1
+        isValidSettingsObject() && content['playModeTestRunnerEnabled'] && content['playModeTestRunnerEnabled'] == 1
     }
 
     boolean isValidSettingsObject() {
-        return content && content.getClass() == LinkedHashMap && content['PlayerSettings']
-    }
-
-    static String stripUnityInstructions(String content) {
-        def lines = content.readLines()
-        lines.collect {
-            if (it.matches(/%TAG !u! tag:unity3d.com,.*:/)) {
-                return ""
-            }
-
-            def m = it =~ /(--- )!u!\d+( &\d+)/
-            if (m) {
-                return "${m[0][1]}${m[0][2]}"
-            }
-
-            return it
-        }
-        .join("\n")
+        return content && content.getClass() == LinkedHashMap
```

Comment:
To me this could also go in `UnityAssetsFile` OR you add a `ProjectSettings`specific test.

Reply:
I thought of removing it completely because as you can see I hoist the inner object already and what class this is I actually don't care. Do you remember why we added the check for `LinkedHashMap`?

Reply:
True, that makes this check obsolet. And i can't remember why we checked for `LinkedhashMap`